edu_edfi_airflow/providers/edfi/transfers/edfi_to_s3.py

```diff
@@ -22,51 +21,68 @@ class EdFiToS3Operator(BaseOperator):
     Use a paged-get to retrieve a particular EdFi resource from the ODS.
     Save the results as JSON lines to `tmp_dir` on the server.
     Once pagination is complete, write the full results to S3.
+
+    Deletes and keyChanges are only used in the bulk operator.
     """
-    template_fields = ('s3_destination_key', 'query_parameters', 'min_change_version', 'max_change_version',)
+    template_fields = (
+        'resource', 'namespace', 'page_size', 'num_retries', 'change_version_step_size', 'query_parameters',
+        's3_destination_key', 's3_destination_dir', 's3_destination_filename',
+        'min_change_version', 'max_change_version', 'enabled_endpoints',
+    )
 
-    @apply_defaults
     def __init__(self,
         edfi_conn_id: str,
-        resource    : str,
+        resource: str,
 
         *,
-        tmp_dir   : str,
+        tmp_dir: str,
         s3_conn_id: str,
-        s3_destination_key: str,
-
-        query_parameters  : Optional[dict] = None,
-        min_change_version: Optional[int ] = None,
-        max_change_version: Optional[int ] = None,
+        s3_destination_key: Optional[str] = None,
+        s3_destination_dir: Optional[str] = None,
+        s3_destination_filename: Optional[str] = None,
+
+        get_deletes: bool = False,
+        get_key_changes: bool = False,
+        min_change_version: Optional[int] = None,
+        max_change_version: Optional[int] = None,
+
+        namespace: str = 'ed-fi',
+        page_size: int = 500,
+        num_retries: int = 5,
         change_version_step_size: int = 50000,
+        query_parameters  : Optional[dict] = None,
 
-        api_namespace  : str = 'ed-fi',  # Should these variables line up 1-to-1 with EdFiClient?
-        api_get_deletes: bool = False,
-        api_retries    : int = 5,
-        page_size      : int = None,
+        enabled_endpoints: Optional[List[str]] = None,
```

Comment:
This is a workaround argument to allow skips to happen immediate in default operators if there are no rows to ingest, instead of skipping after making an API call.